cosmos/operators/_asynchronous/snowflake.py

```diff
@@ -0,0 +1,154 @@
+        **kwargs: Any,
+    ):
+        self.project_dir = project_dir
+        self.profile_config = profile_config
+        self.extra_context = extra_context or {}
+        self.dbt_kwargs = dbt_kwargs or {}
+
+        self.snowflake_conn_id = self.profile_config.profile_mapping.conn_id  # type: ignore
+        self.configuration: dict[str, Any] = {}
+        task_id = self.dbt_kwargs.pop("task_id")
+
+        AbstractDbtLocalBase.__init__(
+            self,
+            task_id=task_id,
+            project_dir=project_dir,
+            profile_config=profile_config,
+            **self.dbt_kwargs,
+        )
+
+        # Configure datasets. This uses the task_id set by AbstractDbtLocalBase
+        if kwargs.get("emit_datasets", True) and settings.enable_dataset_alias and AIRFLOW_VERSION >= Version("2.10"):
+            kwargs = configure_datasets(operator_kwargs=kwargs, task_id=self.task_id)
+
+        super().__init__(
+            snowflake_conn_id=self.snowflake_conn_id,
+            sql=None,  # Setting to None, this will be updated later
+            deferrable=True,
+            **kwargs,
+        )
+
+        self.async_context = extra_context or {}
+        self.async_context["profile_type"] = self.profile_config.get_profile_type()
+        self.async_context["async_operator"] = SnowflakeSqlApiOperator
+
+    @property
+    def base_cmd(self) -> list[str]:
+        return ["run"]
+
+    def execute(self, context: Context, **kwargs: Any) -> None:
+        if settings.enable_setup_async_task:
+            # sql is a required parameter for the SnowflakeSqlApiOperator, so this needs to be set. However, there are
+            # no other required fields. The assumption is that this will be passed in via the connection
+            self.sql = get_remote_sql(async_context=self.async_context, project_dir=self.project_dir)
+            super().execute(context, **kwargs)
+        else:
+            self.build_and_run_cmd(context=context, run_as_async=True, async_context=self.async_context)
+        self._store_template_fields(context=context)
+
+    def execute_complete(self, context: Context, event: dict[str, str | list[str]] | None = None) -> None:
+        # No need to capture the output, it will always be None
+        super().execute_complete(context, event)
+
+        self.log.info("Configuration is %s", str(self.configuration))
+        self._store_template_fields(context=context)
+
+    @provide_session
+    def _store_template_fields(self, context: Context, session: Session = NEW_SESSION) -> None:
+        if not settings.enable_setup_async_task:
+            self.log.info("SQL cannot be made available, skipping registration of compiled_sql template field")
+            return
+
+        sql = get_remote_sql(async_context=self.async_context, project_dir=self.project_dir).strip()
+
+        self.log.debug("Executed SQL is: %s", sql)
+        self.compiled_sql = sql  # TODO: What is this being used for?
```

Comment:
This is a templated field for the class and it gets showed in the UI (task details rendered fields section) for the users to see what SQL query was being executed as part of the task run